app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java

```diff
@@ -3638,15 +3638,21 @@ private int Enrol(int oFamilyId, int oInsureeId, int oPolicyId, int oPremiumId,
                         familyObj.put("insurees", tempInsureesArray);
 
                         // Policy + premium
-                        JSONArray tempPoliciesArray = new JSONArray();
 
                         for (int j = 0; j < policiesArray.length(); j++) {
-                            tempPoliciesArray = policiesArray;
-                            JSONObject premiumObj = premiumsArray.getJSONObject(j);
-                            tempPoliciesArray.getJSONObject(j).put("premium", premiumObj);
+                            JSONObject policyObject = policiesArray.getJSONObject(j);
+                            JSONArray policyPremiums = new JSONArray();
+                            String policyId = policyObject.getString("PolicyId");
+                            for(int k = 0; k < premiumsArray.length(); k++) {
+                                JSONObject premiumObject = premiumsArray.getJSONObject(k);
+                                if(policyId.equals(premiumObject.getString("PolicyId"))) {
```

Comment:
please use string utils equals to omit npe problem

Reply:
I added StringUtils to the project, waiting for approval to use Apache licensed libraries.